walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,118 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.*; 
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> mandatoryTimeRanges = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> optionalTimeRanges = new ArrayList<TimeRange>();
+    // if meeting duration is longer than a day, it is not possible to hold meeting
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+        return mandatoryTimeRanges;
+    }
+    // initialize time range with whole day
+    mandatoryTimeRanges.add(TimeRange.WHOLE_DAY);
+    optionalTimeRanges.add(TimeRange.WHOLE_DAY);
+    Collection<String> MandatoryAttendants = request.getAttendees();
+    Collection<String> optionalAttendants = request.getOptionalAttendees();
+    for(Event e: events) {
```

Comment:
Nit: space after "for" and space after "e" (i.e. "for (Event e : events) {") for consistency within the file.